RoT/Protocol/Cerberus_Attestation.md

```diff
@@ -0,0 +1,1637 @@
+| `8`         | `cert_slot`            | Slot number of the certificate chain to use for attestation challenges. |
+| `8`         | `attestation_protocol` | Protocol to use for attestation requests to the component. | <!-- This needs to be an enum. -->
+| `8`         | `type_len`             | Length of the component type string.  |
+| `type_len`  | `type`                 | ASCII string type for the component.  This is not null terminated. |
+| `ALIGN(32)` | `_`                    | Zero padding.                         |
+
+`enum CFM.ComponentDevice.Check`
+| Value  | Name               | Description                              |
+|--------|--------------------|------------------------------------------|
+| `0x00` | `equal`            | Ensure the reported data is equal to the specified value. |
+| `0x01` | `not_equal`        | Ensure the reported data is not equal to the specified value. |
+| `0x02` | `less_than`        | Ensure the reported data is less than the specified value. |
+| `0x03` | `less_or_equal`    | Ensure the reported data is less than or equal to the specified value. |
+| `0x04` | `greater_than`     | Ensure the reported data is greater than the specified value. |
+| `0x05` | `greater_or_equal` | Ensure the reported data is greater than or equal to the specified value. |
+| `0x06` | `string`           | Ensure strings are equal, only valid for Allowable Data and no bitmask should be provided. |
+<!-- Do we really need String compare?  How is that any different than Equal? -->
+<!-- Why can't we use bitmasks with string comparison?  Seems like that could be
+handy to igore parts of a string that don't matter.  But really, I don't
+understand why this is different than 'equal' -->
+
+### PMR Digest Element
+
+The PMR Digest element contains a list of all allowable digests for a single
+PMR.  These digests represent the final measurement reported by the PMR, as
+would be reported by `Challenge` or `Get PMR` requests.
+
+`bitfield CFM.ComponentDevice.PMRDigest`
+| Type                      | Name             | Description                   |
+|---------------------------|------------------|-------------------------------|
+| `8`                       | `pmr_id`         | Identifier for the PMR to attest.  The Cerberus Challenge Protocol allows this to be between 0 and 4. |
+| `8`                       | `pmr_data_count` | Number of measurement entries used to compute this PMR. | <!-- I'm not sure why we need this.  It also could change between versions. -->
+| `00000b`                  | `_`              | Padding for the unused HashType bits.  This must be 0. |
+| `Manifest.HashType`       | `hash_type`      | The type of hash used to generate the PMR digest. |
+| `8`                       | `digest_count`   | Number of allowable digests for this PMR. |
+| `hash_type(digest_count)` | `pmr_digest`     | Expected digest for the PMR.  |
```

Comment:
Presumably this the actual value currently stored in the PMR, insofar as the PMR is built by repeated hashing?